src/hooks/operators.ts

```diff
@@ -0,0 +1,179 @@
+                } catch (e) {
+                    if (!(e instanceof z.ZodError)) {
+                        throw e
+                    }
+
+                    console.warn('Failed to parse an operator', operator, e)
+                }
+            }
+
+            return null
+        },
+    })
+}
+
+export function useOperatorForWallet(address = '') {
+    return useOperatorForWalletQuery(address).data || null
+}
+
+export function useIsLoadingOperatorForWallet(address = ''): boolean {
+    const query = useOperatorForWalletQuery(address)
+
+    return query.isLoading || query.isFetching
+}
+
+export function useOperatorStatsForWallet(address = '') {
+    const operator = useOperatorForWallet(address)
+
+    if (!operator) {
+        return operator
+    }
+
+    const {
+        delegatorCount: numOfDelegators,
+        valueWithoutEarnings: value,
+        stakes,
+    } = operator
+
+    return {
+        numOfDelegators,
+        numOfSponsorships: stakes.length,
+        value,
+    }
+}
+
+export function useDelegacyStats(address = '') {
+    const [stats, setStats] = useState<DelegacyStats | undefined | null>()
+
+    const addr = address.toLowerCase()
+
+    useEffect(() => {
+        let mounted = true
+
+        if (!addr) {
+            setStats(null)
+
+            return () => {}
+        }
+
+        setStats(undefined)
+
+        setTimeout(async () => {
+            const operators = await getDelegacyForWallet(addr, {
+                batchSize: 1000,
+                onBeforeComplete(total, parsed) {
+                    if (total !== parsed) {
+                        errorToast({
+                            title: 'Warning',
+                            desc: `Delegation stats are calculated using ${parsed} out of ${total} available operators due to parsing issues.`,
+                        })
+                    }
+                },
+            })
+
+            if (!mounted) {
+                return
+            }
+
+            if (!operators.length) {
+                return void setStats({
+                    value: toBN(0),
+                    minApy: 0,
+                    maxApy: 0,
+                    numOfOperators: 0,
+                })
+            }
+
+            let minApy = Number.POSITIVE_INFINITY
+
+            let maxApy = Number.NEGATIVE_INFINITY
+
+            operators.forEach(({ apy }) => {
+                minApy = Math.min(minApy, apy)
+
+                maxApy = Math.max(maxApy, apy)
+            })
+
+            const value = operators.reduce(
+                (sum, { myShare }) => sum.plus(myShare),
+                toBN(0),
+            )
+
+            setStats({
+                value,
+                minApy,
+                maxApy,
+                numOfOperators: operators.length,
+            })
+        })
+
+        return () => {
+            mounted = false
+        }
+    }, [addr])
+
+    return stats
+}
+
+export function useDelegacyForWalletQuery(
```

Comment:
Delegacy? Maybe let's use Delegations to keep internal language consistency?

Reply:
Hm, maybe maybe.. It is true that it is kind of a new word. Delegacy = body of delegation collective. If it got you cornered a bit into thinking wtf than maybe it is me trying too hard after all. I'll go back to plurals.